src/engraving/rw/read400/readcontext.cpp

```diff
@@ -506,6 +506,55 @@ void ReadContext::reconnectBrokenConnectors()
     doReconnectBrokenConnectors();
 }
 
+void ReadContext::clearOrphanedConnectors()
+{
+    if (_connectors.empty() && _pendingConnectors.empty()) {
+        return;
+    }
+
+    LOGD("XmlReader::~XmlReader: there are unpaired connectors left");
+
+    std::set<LinkedObjects*> deletedLinks;
+
+    auto deleteConnectors = [&deletedLinks](std::shared_ptr<ConnectorInfoReader> c) {
+        EngravingItem* conn = c->releaseConnector();
+
+        LinkedObjects* links = conn->links();
+        bool linksWillBeDeleted = links && links->size() == 1;
+
+        if (conn && !conn->isTuplet()) {     // tuplets are added to score even when not finished
+            if (linksWillBeDeleted) {
+                deletedLinks.insert(links);
+            }
+
+            delete conn;
+        }
+    };
+
+    if (!_connectors.empty()) {
+        for (auto& c : _connectors) {
+            deleteConnectors(c);
+        }
+        _connectors.clear();
+    }
+
+    if (!_pendingConnectors.empty()) {
+        for (auto& c : _pendingConnectors) {
+            deleteConnectors(c);
+        }
+        _pendingConnectors.clear();
+    }
+
+    for (auto& it : m_staffLinkedElements) {
+        for (std::pair<LinkedObjects*, Location>& pair : it.second) {
+            LinkedObjects*& link = pair.first;
+            if (deletedLinks.count(link)) {
+                link = nullptr;
```

Comment:
Is it safe to have nullptrs there? Shouldn't you rather erase this pair from the list? For example, `ReadContext::doAddLink` seems to be not very careful about whether `….first` might be nullptr.

Reply:
Good point. Thanks